src/routes/_surfaces/assets/TableAssets.svelte

```diff
@@ -70,40 +133,45 @@
             profile={derivedTable?.profile ?? []}
             head={derivedTable?.preview ?? []}
             sizeInBytes={derivedTable?.sizeInBytes ?? 0}
-            on:query={async () => {
-              // check existing models to avoid a name conflict
-              const existingNames = $persistentModelStore?.entities
-                .filter((model) => model.name.includes(`query_${tableName}`))
-                .map((model) => model.tableName)
-                .sort();
-              const nextName =
-                existingNames.length === 0
-                  ? `query_${tableName}`
-                  : `query_${tableName}_${existingNames.length + 1}`;
-
-              const response = await dataModelerService.dispatch("addModel", [
-                {
-                  name: nextName,
-                  query: `select * from ${tableName}`,
-                },
-              ]);
-
-              // change the active asset to the new model
-              await dataModelerService.dispatch("setActiveAsset", [
-                EntityType.Model,
-                response.id,
-              ]);
-
-              notificationStore.send({
-                message: `queried ${tableName} in workspace`,
-              });
-            }}
-            on:delete={() => {
-              dataModelerService.dispatch("dropTable", [tableName]);
-            }}
+            on:rename={() => renameHandler(tableName, id)}
+            on:query={() => queryHandler(tableName)}
+            on:delete={() =>
+              dataModelerService.dispatch("dropTable", [tableName])}
           />
         </div>
       {/each}
     {/if}
   </div>
+  <Modal
```

Comment:
We should move these to a separate component. This component is getting a little too big.

Reply:
Yep, good call, made the change 👍